### DISCUSSION TITLE
How to register a scalar type with static newScalar()?

### DISCUSSION THREAD

**Discussion by mxmlnglt** [graphql-java-kickstart/graphql-spring-boot#460] [Q&A] [answered]
I'd like to register a scalar type the *correct* way.

Indeed I found this issue https://github.com/graphql-java-kickstart/graphql-spring-boot/issues/151 which has both a link to the [dedicated wiki page](https://github.com/graphql-java-kickstart/graphql-spring-boot/wiki/Creating-a-custom-scalar-type) and a [GraphQLDate example](https://github.com/donbeave/graphql-java-datetime/blob/master/graphql-java-datetime/src/main/java/com/zhokhov/graphql/datetime/GraphQLDate.java).

However these classes are written the old way i.e. using the `super()` constructor... which is now [`@Deprecated`](https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/schema/GraphQLScalarType.java#L61).

How can I use the *new* way by using the static method `newScalar()`?

Thanks.

**Comment by pstarritt-evooq** [comment]
Then expose the object as bean

```
GraphQLScalarType.newScalar()
        .name("LocalDate")
        .description("Description")
        .coercing(new Coercing<Date, String>() {

          @Override
          public String serialize(Object dataFetcherResult) throws CoercingSerializeException {
            // stuff
            return null; 
          }

          @Override
          public Date parseValue(Object input) throws CoercingParseValueException {
               // stuff
            return null;
          }

          @Override
          public Date parseLiteral(Object input) throws CoercingParseLiteralException {
           // stuff
            return null;
          }
        })
        .build();

**Comment by mxmlnglt** [comment]
> Then expose the object as bean

could you be more explicit? thanks.

**Comment by pstarritt-evooq** [comment]
After you create your new scalar object, you should make it a spring bean which will register it with graphql

**Comment by mxmlnglt** [comment]
> After you create your new scalar object, you should make it a spring bean which will register it with graphql

OK... so no need to call `.coercion(..)` then?
I'm sorry but I don't really understand.

**Comment by oliemansm** [comment, marked as answer]
@mxmlnglt 
In one of your Spring `@Configuration` files add something like this (using your own scalar implementation of course):
```java
  @Bean
  public GraphQLScalarType longScalar() {
    return GraphQLScalarType.newScalar()
        .name("Long")
        .description("Long type")
        .coercing(new LongCoercing())
        .build();
  }
```
This creates and exposes a Spring Bean which is picked up automatically by GraphQL.
